src/devices/bus/ieee488/remote488.cpp

```diff
@@ -0,0 +1,721 @@
+    the accompanying byte, according to this table.
+
+    | Bit | Signal |
+    |-----+--------|
+    | 0   | ATN    |
+    | 1   | IFC    |
+    | 2   | REN    |
+    | 3   | SRQ    |
+
+    A "R" message clears (sets to 0, i.e. asserts) all signals that
+    are set to "1" in the byte, all others retain their value.
+    A "S" message sets signals to 1, i.e. de-asserts them.
+    The remotizer may respond with a "R" message if an incoming "S"
+    message is attempting to set to 1 a signal that is locally forced
+    to 0.
+
+    Limits & potential issues:
+    - There are no checks for violations of IEEE-488 protocol. It's
+      possible, for example, to put the bus in an illegal state by
+      sending a 'E' msg when the local controller is asserting ATN
+      (thus creating a phantom parallel poll).
+    - The unacknowledged nature of D/E msgs may cause problems if
+      the sender of the msgs must be synchronized byte-for-byte with
+      the receiver. There is no direct way, for example, for the
+      sender of a string of bytes to know how many have been accepted
+      by the receiver.
+    - It's difficult to achieve accurate synchronization between the
+      local emulation time and the external time.
+
+    TODOs/possible enhancements:
+    - Implement handling of incoming Q msgs (needed when parallel poll
+      is being performed by a remote controller)
+    - Enhancement: implement a msg for accurate time synchronization
+
+*********************************************************************/
+
+#include "emu.h"
+#include "remote488.h"
+
+// Debugging
+#define VERBOSE 0
+#include "logmacro.h"
+
+// Bit manipulation
+namespace {
+	template<typename T> constexpr T BIT_MASK(unsigned n)
+	{
+		return (T)1U << n;
+	}
+
+	template<typename T> void BIT_CLR(T& w , unsigned n)
+	{
+		w &= ~BIT_MASK<T>(n);
+	}
+
+	template<typename T> void BIT_SET(T& w , unsigned n)
+	{
+		w |= BIT_MASK<T>(n);
+	}
+
+	template<typename T> void COPY_BIT(bool bit , T& w , unsigned n)
+	{
+		if (bit) {
+			BIT_SET(w , n);
+		} else {
+			BIT_CLR(w , n);
+		}
+	}
+}
+
+// Message types
+constexpr char MSG_SIGNAL_CLEAR  = 'R'; // I/O: Clear signal(s)
+constexpr char MSG_SIGNAL_SET    = 'S'; // I/O: Set signal(s)
+constexpr char MSG_DATA_BYTE     = 'D'; // I/O: Cmd/data byte (no EOI)
+constexpr char MSG_END_BYTE      = 'E'; // I/O: Data byte (with EOI)
+constexpr char MSG_PP_DATA       = 'P'; // I:   Parallel poll data
+constexpr char MSG_PP_REQUEST    = 'Q'; // O:   Request PP data
+constexpr char MSG_ECHO_REQ      = 'J'; // O:   Heartbeat msg: echo request
+constexpr char MSG_ECHO_REPLY    = 'K'; // I:   Heartbeat msg: echo reply
+
+// Timings
+constexpr unsigned POLL_PERIOD_US   = 20;   // Poll period (µs)
+constexpr unsigned HEARTBEAT_MS     = 500;  // Heartbeat ping period (ms)
+constexpr unsigned MAX_MISSED_HB    = 3;    // Missed heartbeats to declare the connection dead
+
+// Timers
+enum {
+	TMR_ID_POLL,
+	TMR_ID_HEARTBEAT
```

Comment:
Please put this in the device class at public level.  Device timer IDs can't conflict across a device and all derived devices and mixins, so it's good practice to have them declared in the device class.